### PR TITLE
Optimize xTaskIncrementTick for configNUMBER_OF_CORES > 1

### PR DESCRIPTION
**Pull request by cymizer**
Description
-----------

The original implementation only initializes the first variable. After executing xTaskIncrement, the schedule might not behave as expected.

Test Steps
-----------
N/A

Checklist:
----------

- [x] I have tested my changes. No regression in existing tests.
- [ ] I have modified and/or added unit-tests to cover the code changes in this Pull Request.

Related Issue
-----------
N/A

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: tasks.c [hunk 1]
```diff
@@ -4699,7 +4699,7 @@ BaseType_t xTaskIncrementTick( void )
     BaseType_t xSwitchRequired = pdFALSE;
 
     #if ( configUSE_PREEMPTION == 1 ) && ( configNUMBER_OF_CORES > 1 )
-    BaseType_t xYieldRequiredForCore[ configNUMBER_OF_CORES ] = { pdFALSE };
+    BaseType_t xYieldRequiredForCore[ configNUMBER_OF_CORES ] = { [ 0 ... configNUMBER_OF_CORES ] = pdFALSE };
```
[the remark this answers is not in the document]
Comment (jefftenney) [reply tasks.c:4702]: There is an [existing loop](https://github.com/FreeRTOS/FreeRTOS-Kernel/blob/bb7056baef862d39b92394622b6cd2224a307b72/tasks.c#L4860) where this change would fit nicely.

Existing Code:
```
for( xCoreID = 0; xCoreID < ( ( BaseType_t ) configNUMBER_OF_CORES ); xCoreID++ )
{
    if( listCURRENT_LIST_LENGTH( &( pxReadyTasksLists[ pxCurrentTCBs[ xCoreID ]->uxPriority ] ) ) > 1U )
    {
        xYieldRequiredForCore[ xCoreID ] = pdTRUE;
    }
    else
    {
        mtCOVERAGE_TEST_MARKER();
    }
}
```

Proposed Code:
```
for( xCoreID = 0; xCoreID < ( ( BaseType_t ) configNUMBER_OF_CORES ); xCoreID++ )
{
    if( listCURRENT_LIST_LENGTH( &( pxReadyTasksLists[ pxCurrentTCBs[ xCoreID ]->uxPriority ] ) ) > 1U )
    {
        xYieldRequiredForCore[ xCoreID ] = pdTRUE;
    }
    else
    {
        xYieldRequiredForCore[ xCoreID ] = pdFALSE;
    }
}
```

[the remark this answers is not in the document]
Comment (cymizer) [reply tasks.c]: hi @aggarg , @jefftenney 
Thank you for your valuable suggestions. I have made the changes based on your recommendations. Kindly assist with the review.

File: tasks.c [hunk 2]
```diff
@@ -4905,7 +4905,11 @@ BaseType_t xTaskIncrementTick( void )
                         if( pxCurrentTCBs[ xCoreID ]->xPreemptionDisable == pdFALSE )
                     #endif
                     {
-                        if( ( xYieldRequiredForCore[ xCoreID ] != pdFALSE ) || ( xYieldPendings[ xCoreID ] != pdFALSE ) )
+                        #if ( configUSE_TIME_SLICING == 1 )
+                            if( ( xYieldRequiredForCore[ xCoreID ] != pdFALSE ) || ( xYieldPendings[ xCoreID ] != pdFALSE ) )
+                        #else
+                            if( xYieldPendings[ xCoreID ] != pdFALSE )
+                        #endif /* #if ( configUSE_TIME_SLICING == 1 ) */
```
Comment (jefftenney) [inline tasks.c:4912]: This looks fine to me now, but can't help but wonder if we should eliminate `xYieldRequiredForCore` altogether.  (Sorry for not seeing this earlier.)  The code above that sets `xYieldRequiredForCore[ xCoreID ]` would just set xYieldPendings[ xCoreID ] instead, leaving the original test-coverage marker in the "else" case.  Would you mind investigating that possibility?  Any concerns @aggarg?

**Review by jefftenney — APPROVED** [review]
Looks good.  This PR resolves the issue of the uninitialized array, and it slightly increases performance too.  The only functional difference I see is for a task with preemption disabled (via `vTaskPreemptionDisable()`).  For that task, a time-slicing yield is dropped before this PR and is delayed after this PR.  We could see that as an improvement too.

**Comment by aggarg** [comment]
>  For that task, a time-slicing yield is dropped before this PR and is delayed after this PR. 

Thanks for pointing that. It should be okay as the scheduler will not evict a task for which the preemption is disabled - https://github.com/FreeRTOS/FreeRTOS-Kernel/blob/main/tasks.c#L1230.

**Review by aggarg — APPROVED** [review]

**Review by kar-rahul-aws — APPROVED** [review]

**Review by chinglee-iot — APPROVED** [review]

**Pull request merged**